### PR TITLE
OO-#64 adding the condition, in case of today, active

### PR DESCRIPTION
**Pull request by mcmpp**

### CODE REVIEW

File: src/main/scala/ch/openolitor/stammdaten/models/AbosModels.scala [hunk 1]
```diff
@@ -35,7 +35,7 @@ case class AboId(id: Long) extends BaseId
 object IAbo {
   def calculateAktiv(start: LocalDate, ende: Option[LocalDate]): Boolean = {
     val yesterday = LocalDate.now.minusDays(1)
-    start.isBefore(LocalDate.now) && (ende map (_.isAfter(yesterday)) getOrElse true)
+    (start.isBefore(LocalDate.now) || start.isEqual(LocalDate.now)) && (ende map (_.isAfter(yesterday)) getOrElse true)
```
Comment (AlwinEgger) [inline src/main/scala/ch/openolitor/stammdaten/models/AbosModels.scala]: Isn't this comparing by milliseconds? Does this wok in all cases of start (one minute past midnight and one minute before midnight)?

File: src/main/scala/ch/openolitor/stammdaten/models/AbosModels.scala [hunk 2]
```diff
@@ -35,7 +35,7 @@ case class AboId(id: Long) extends BaseId
 object IAbo {
   def calculateAktiv(start: LocalDate, ende: Option[LocalDate]): Boolean = {
     val yesterday = LocalDate.now.minusDays(1)
-    start.isBefore(LocalDate.now) && (ende map (_.isAfter(yesterday)) getOrElse true)
```
Comment (fabianpage) [inline src/main/scala/ch/openolitor/stammdaten/models/AbosModels.scala:38]: couldn't you replace the first part with:
!start.isAfter(now)

**Review by fabianpage — APPROVED** [review]

File: src/main/scala/ch/openolitor/stammdaten/models/AbosModels.scala (hunk 2 above)
Comment (mcmpp) [reply src/main/scala/ch/openolitor/stammdaten/models/AbosModels.scala:38]: done

File: src/main/scala/ch/openolitor/stammdaten/models/AbosModels.scala [hunk 3]
```diff
@@ -35,7 +35,7 @@ case class AboId(id: Long) extends BaseId
 object IAbo {
   def calculateAktiv(start: LocalDate, ende: Option[LocalDate]): Boolean = {
     val yesterday = LocalDate.now.minusDays(1)
-    (start.isBefore(LocalDate.now) || start.isEqual(LocalDate.now)) && (ende map (_.isAfter(yesterday)) getOrElse true)
+    !start.isAfter(LocalDate.now) && (ende map (_.isAfter(yesterday)) getOrElse true)
```
Comment (AlwinEgger) [inline src/main/scala/ch/openolitor/stammdaten/models/AbosModels.scala]: This still returns different results if we execute in the morning or afternoon if we try with a date XX.YY.2018 12:00.
BTW: This case can very easily be covered with a unit test.

Comment (fabianpage) [reply src/main/scala/ch/openolitor/stammdaten/models/AbosModels.scala]: You could compare with daysBetween (http://www.joda.org/joda-time/apidocs/org/joda/time/Days.html#daysBetween-org.joda.time.ReadableInstant-org.joda.time.ReadableInstant-).

File: src/main/scala/ch/openolitor/stammdaten/models/AbosModels.scala (hunk 1 above)
Comment (mcmpp) [reply src/main/scala/ch/openolitor/stammdaten/models/AbosModels.scala]: I have added a test to check the midnight midday cases using calculateAktiv

Comment (AlwinEgger) [reply src/main/scala/ch/openolitor/stammdaten/models/AbosModels.scala]: Ok, great

**Review by AlwinEgger — APPROVED** [review]

**Pull request merged**